### ISSUE TITLE
Escape characters in string literals cause line truncation in RemedyBG's text viewer and sometimes crashes

### ISSUE THREAD

**Issue by ebeachly** [x13pixels/remedybg-issues#300]
Hi, thank you for adding comment/keyword highlighting! It's super nice! I think I found a bug related to it though.

I'm using RemedyBG 0.3.9.8

I noticed that escape characters in string literals cause RemedyBG to not display the full line of text, and sometimes cause it to crash. There's no problem if the string literal is in a comment.

Paste the following code block into a new file and open it in RemedyBG. My text editor uses windows-style line endings.

```
//Open this file in RemedyBG. You don't need to actually compile or run it.
int main(void) {

	const char a[] = "\"";

	const char b[] = "\"\"";

	//The semicolon on the following line isn't displayed in RemedyBG
	const char c[] = "\"\"\"";

	//Notice that if the string is commented out, RemedyBG doesn't have a problem
	//const char c[] = "\"\"\"";

	//The last quotation mark and semicolon on the following line isn't displayed in RemedyBG
	const char d[] = "\"\"\"\"";

	//RemedyBG crashed on me once when the following line was uncommented, otherwise it truncates the last three characters
	//const char e[] = "\"\"\"\"\"";

	const char f[] = "\"\\";

	//The semicolon on the following line isn't displayed in RemedyBG
	const char g[] = "\"\"\\";

	//RemedyBG crashed on me once when the following line was uncommented, otherwise it truncates the last two characters
	//const char h[] = "\"\"\"\\";

	return a[0];
}
```

Here's what it looks like in RemedyBG:

![StringEscapeBug](https://github.com/x13pixels/remedybg-issues/assets/5232113/45adc85d-8973-4bc2-a28b-8079880b8cf4)

If I save the file with Unix-style line endings, an extra character is truncated.

This happens if I disable comment/keyword highlighting as well.

**Comment by x13pixels** [comment]
Thanks for the detailed writeup! I'll get this fixed.

**Comment by x13pixels** [comment]
Fixed for next release 0.3.9.9. Thanks again for the report.
